### ISSUE TITLE
[BUG] Black screen output as of commit 029636d

### ISSUE THREAD

**Issue by sgsdxzy** [kijai/ComfyUI-HunyuanVideoWrapper#99]
GPU: 2080Ti hard-modded to 22GB
Environment: Linux x86-64, torch 2.5.1 + cuda 12.4, diffusers 0.31.0, transformers 4.46.3, ComfyUI latest
Status: workflow `examples/hyvideo_lowvram_blockswap_test.json` runs fine, but the generated video is all black. Can confirm the same workflow works correctly under commit 6655642ea6b915cf94ff0e2037aa7e260f9e7d26.

Workflow screenshot:
![image](https://github.com/user-attachments/assets/d7b5e97b-a0ea-448f-8f50-0d1cb7a3f63e)

Notes:
1. The `RuntimeWarning: invalid value encountered in cast` in the console log looks suspicous: 
```
got prompt
llm prompt attention_mask shape: torch.Size([1, 256]), masked tokens: 21
clipL prompt attention_mask shape: torch.Size([1, 77]), masked tokens: 22
Input (height, width, video_length) = (320, 480, 5)
Swapping 20 double blocks and 0 single blocks
Sampling 5 frames in 2 latents at 480x320 with 30 inference steps
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:55<00:00,  1.85s/it]
Allocated memory: memory=5.964 GB
Max allocated memory: max_memory=6.509 GB
Max reserved memory: max_reserved=6.750 GB
Decoding rows: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00,  3.44it/s]
Blending tiles: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 15.91it/s]
/home/samba/ssd/ComfyUI/custom_nodes/ComfyUI-VideoHelperSuite/videohelpersuite/nodes.py:104: RuntimeWarning: invalid value encountered in cast
  return tensor_to_int(tensor, 8).astype(np.uint8)
Prompt executed in 65.20 seconds
```
2. compared with the working commit of 6655642ea6b915cf94ff0e2037aa7e260f9e7d26, the latest commit 029636d2a79986afa74577d1a633a4e370bd6206 runs much faster and uses a lot less VRAM.

I can do a bisect to find the commit that breaks if you need.

**Comment by kijai** [comment]
The model doesn't seem to work on fp16.

**Comment by Smiden3** [comment]
https://github.com/kijai/ComfyUI-HunyuanVideoWrapper/issues/94

**Comment by sgsdxzy** [comment]
commit a01b0796be154ee224caa15d0709d1d5f6a51fe8 seems to break it.
As 2080Ti only has fp16, let me try if setting the model to bf16 works at all...

**Comment by sgsdxzy** [comment]
Setting `base_precision` in `HunyuanVideo Model Loader` to `bf16` fixes the black screen. The generation result is different from previous versions under the same random number seed, though.

**Comment by kijai** [comment]
> Setting `base_precision` in `HunyuanVideo Model Loader` to `bf16` fixes the black screen. The generation result is different from previous versions under the same random number seed, though.

That is very well possible, as this is still under development the noise and such might change making it hard to reproduce earlier version results exactly.

**Comment by RageshAntonyHM** [comment]
@kijai 

I set to bf16 already, still gets black screen. Pls help me

<img width="1280" alt="image" src="https://github.com/user-attachments/assets/9e4b38eb-9be8-4ec9-ab61-52bd0228fb48">

**Comment by testingNetqa** [comment]
same here, progress without warning, but the output and the content in the .mp4 files still black

**Comment by RageshAntonyHM** [comment]
@kijai 
Well,  Ran a script and got like this:

Torch version: 2.4.1+cu124
CUDA version: 12.4

**Comment by kijai** [comment]
> @kijai 
> Well,  Ran a script and got like this:
> 
> Torch version: 2.4.1+cu124
> CUDA version: 12.4

That does explain it, you need to update to 2.5.1+cu124

**Comment by RageshAntonyHM** [comment]
@kijai 

Yeah. After upgrading it's working. Thanks.

<img width="1280" alt="image" src="https://github.com/user-attachments/assets/338e6c5d-38d6-4793-9f7c-ca8a8dd95358">
